base/logging.jl

```diff
@@ -341,8 +342,10 @@ end
         # Try really hard to get the message to the logger, with
         # progressively less information.
         try
-            msg = "Exception while generating log record in module $_module at $filepath:$line"
-            handle_message(logger, Error, msg, _module, group, id, filepath, line; exception=err)
+            msg = "Exception while generating log record:\n"
+            str = join("  " .* split(sprint(showerror, err), "\n"), "\n")
+            msg *= str
+            handle_message(logger, Error, msg, _module, group, id, filepath, line)
```

Comment:
I don't think this should be done inside `dispatch_message` as that will bake in display logic in a way which can't be overridden.  Rather, I think it should be part of the logger. See for example https://github.com/JuliaLang/julia/pull/25136/files#diff-f1d5a62767988822d6e6fd5db69f5ea9R79